modules/ROOT/pages/sample-data-java-sdk.adoc

```diff
@@ -0,0 +1,472 @@
+
+ /**
+  * Indicates that Sample Data resolution was attempted on a component which doesn't support that capability
+  */
+ public final static String NOT_SUPPORTED = "NOT_SUPPORTED";
+
+ /**
+  * Indicates that Sample Data resolution was attempted on an unregistered Extension
+  */
+ public final static String INVALID_TARGET_EXTENSION = "INVALID_TARGET_EXTENSION";
+
+ private final String failureCode;
+
+ public SampleDataException(String message, String failureCode) {
+   super(createStaticMessage(message));
+   this.failureCode = failureCode;
+ }
+
+ public SampleDataException(String message, String failureCode, Throwable cause) {
+   super(createStaticMessage(message), cause);
+   this.failureCode = failureCode;
+ }
+
+ public SampleDataException(I18nMessage message, String failureCode) {
+   super(message);
+   this.failureCode = failureCode;
+ }
+
+ public SampleDataException(I18nMessage message, String failureCode, Throwable cause) {
+   super(message, cause);
+   this.failureCode = failureCode;
+ }
+
+ /**
+  * @return The failure code of the error that produced the error
+  */
+ public String getFailureCode() {
+   return failureCode;
+ }
+}
+----
+
+=== @SampleData
+
+This annotation associates a resolver to the enhanced component. It is
+applied at the type level in sources and at the method level for operations.
+
+The following example illustrates a `@SampleData` annotation:
+
+[source,java,linenums]
+----
+/**
+* Associates a {@link Source} class or operation method with a {@link SampleDataProvider}.
+*
+* @since 1.4.0
+* @see SampleDataProvider
+*/
+@Target({TYPE, METHOD})
+@Retention(RUNTIME)
+@Documented
+public @interface SampleData {
+
+ /**
+  * @return the associated {@link SampleDataProvider}
+  */
+ Class<? extends SampleDataProvider> value();
+
+ /**
+  * @return bindings between parameters in the {@link SampleData#value()} to extraction expressions.
+  */
+ Binding[] bindings() default {};
+
+}
+----
+
+== Sample Data Examples
+
+=== Using Only Operation Parameters
+
+Use this example to provide static sample data without using a connection.
+
+Imagine there is a new GitHub connector that contains an operation that
+retrieves information for a given user in JSON format. However, you do not want
+to consume an API call while providing sample data because there are a limited
+number of calls or because it is too costly. Instead, directly implement
+the operation.
+
+The following example illustrates a definition of the operation with the provider attached
+through the `@SampleData` annotation:
+
+[source,java,linenums]
+----
+@SampleData(JsonSampleDataProvider.class)
+public Result<InputStream, Void> getUserInfo(String username, @Connection GitHubConnection connection) {
+ // operation code...
+}
+----
+
+The following example illustrates the sample data provider implementation, which uses the same
+mechanism to inject parameters. The `@Parameter` annotation has the same name
```

Comment:
To what does "the sample data provider implementation" refer: the implementation shown previous to this?
Is that what "the same mechanism" also refers to?